tuf/scripts/repo.py

```diff
@@ -514,19 +526,11 @@ def add_verification_key(parsed_arguments):
 
     if parsed_arguments.role not in ('root', 'targets', 'snapshot', 'timestamp'):
       raise exceptions.Error('The given --role is not a top-level role.')
-
-    elif parsed_arguments.role == 'root':
-      repository.root.add_verification_key(imported_pubkey)
-
-    elif parsed_arguments.role == 'targets':
-      repository.targets.add_verification_key(imported_pubkey)
-
-    elif parsed_arguments.role == 'snapshot':
-      repository.snapshot.add_verification_key(imported_pubkey)
-
-    # The timestamp role..
     else:
-      repository.timestamp.add_verification_key(imported_pubkey)
+      role = getattr(repository, parsed_arguments.role)
+      role.add_verification_key(imported_pubkey)
+      if role.threshold != parsed_arguments.threshold:
+        role.threshold = parsed_arguments.threshold
```

Comment:
I believe if `--threshold` is not set, this resets role.threshold to 1.

Reply:
Yep exactly, so by default it will always be a threshold of 1, but if people want to change they have to specify the threshold each time they trust keys

Reply:
That is extremely surprising and makes it easy to create a security disaster: If I do
`repo.py --trust --pubkeys key1.pub key2.pub key3.pub --role root`
I don't expect my root threshold to just be reset to 1 from whatever it was before, yet I think that's what happens now.

I think there's two choices:
* no `--threshold` leads to not modifying threshold here. This might mean you'd have to change the default value for the argument and then change how `delegate()` handles the default
* make `--threshold` a required argument with `--trust`

Reply:
Good point, will update that

Reply:
So I kept the same behavior than before for delegate() default of 1 if not set.

For trust it will only be updated if the threshold key is set, otherwise it will keep the last value